bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/bridge.xml

```diff
@@ -62,4 +62,44 @@
 		</config-description>
 	</bridge-type>
 
+	<!-- CLIP 2 Bridge -->
+	<bridge-type id="clip2">
```

Comment:
Since CLIP 2 is the technical abbreviation of the implemented protocol, I don't think this alone is the best id for the bridge type. On the other hand, we need to distinguish the two different protocols offered by the same physical Hue bridge. Perhaps `bridge-clip2` or `bridge-v2`?

Reply:
Perhaps simply ‘bridge2’ ??

Reply:
That leaves some uncertainty about Bridge v2 vs. API v2 (running on Bridge v2) - which also disqualifies my "bridge-v2" proposition. So probably either `bridge-clip2`, `bridge-api-v2` or `bridge-api2` would be better?

Reply:
I don't remember ever stating that. I have on some occasions referenced https://www.openhab.org/docs/developer/guidelines.html#java-coding-style stating that **channel ids** should be `lowerCamelCase`.

However, things might change soon as it appears that we haven't really had a formal naming convention - see https://github.com/openhab/openhab-addons/pull/12231#discussion_r1128602473.

On short term: There is no naming convention yet, but some conventions have gained higher adoption than others, so I can still provide some recommendations. I would expect a naming convention soon, so we will have cross-binding consistency going forward. My current recommendation would be `with-hyphens` for thing type ids as well as channel ids. `lowerCamelCase` is also acceptable, but I would not recommend other conventions like `with_underscores` or `noseparatorsatall`.

Reply:
> My current recommendation would be with-hyphens for .. channel ids.

Umm. Can I politely mention that in your earlier review, above, your told me to change from hyphens to camelCase; which I did already; so are you now asking me to revert that change?

Reply:
> > My current recommendation would be with-hyphens for .. channel ids.

> in your earlier review, above, your told me to change from hyphens

To be accurate, it was from underscores, not from hyphens.

> to camelCase; which I did already

There is no contradiction here:

> I have on some occasions referenced https://www.openhab.org/docs/developer/guidelines.html#java-coding-style stating that channel ids should be lowerCamelCase.

So yes, I told you that, because it was documented and I was not aware that this documented guideline had not been agreed upon.

> so are you now asking me to revert that change?

No, as stated:

> My current recommendation would be with-hyphens for thing type ids as well as channel ids. lowerCamelCase is also acceptable

So current naming is accepted, but it is no longer my personal recommendation/advise. The reason for that:
- Consistency with system channel type ids (e.g. `system.battery-level`).
- UID is completely in lower case.

If there will be no arguments against this convention, and if we are going to have a convention, I would therefore consider this as a good candidate. But since I'm not a fortune teller, for now this is just my personal recommendation, and you can choose to ignore it if you don't see any advantage is renaming (again).

If there will be a convention, it will only apply to new bindings. For existing bindings, usually the existing binding convention should be used for internal consistency, i.e. to not have mixed conventions. For major rewrites/additions (like Netatmo last year and Hue this year), IMHO would be a good moment to migrate. For the Hue binding having all new bridge and thing types with new channels, I see no benefit on keeping the old `lower_case_underscore` convention.

Reply:
Ok. I will wait before changing anything for the time being.

Reply:
We have now finally agreed on an official naming convention: openhab/openhab-docs#2061. I would expect this page to be updated tomorrow: https://next.openhab.org/docs/developer/guidelines.html

So when breaking from the previous Hue binding naming convention, we should adopt this one, i.e. `lower-case-hyphen` for channel ids. This is not a strict requirement since it only applies to new bindings, but it would be a bit strange to change the naming convention from A to B when actually C is the official one.

Sorry for the inconvenience, but the discussion and agreement about the naming convention happened during the last three weeks. FYI, I have taken my own medicine: 26a4447787c8446dd0d3b3a7fb0d0a1ee5c3fc0c 🙂

Reply:
> finally agreed upon an official naming convention

@jlaur thanks for the update. So to get back to the original issue, I suppose the bridge thing shall be `bridge2` (shorter, so my preference) or `bridge-api2` (probably yours). Or??

Reply:
As I put in third comment (https://github.com/openhab/openhab-addons/pull/13570#discussion_r1148825274) this is ambiguous because it could either mean Hue Bridge 2.0 or API v2.

Reply:
> this is ambiguous 

IMHO that is the purpose of documentation. But you just tell me what you want please.

Reply:
I don't have any strong opinions here, but my proposal would be `bridge-api2`. This fixes the original issue I saw with the bridge name being only a technical API abbreviation - by combining the physical element (the Hue Bridge) with the specific API version (API v2 or CLIP 2) while also being explicit about the API version without confusing this with the Hue Bridge version.

Reply:
Ok.